scene.js

```diff
@@ -72,31 +73,41 @@ class Scene {
         mat4.multiply(this.viewMatrix, yRotation, xRotation);
         this.viewMatrix[14] = -translate;
 
-        var firstNode = this.nodes[0];
-
-        drawNodeRecursive(this, firstNode, mat4.create());
+        // draw the scene; note - do not assume a single fixed root
+        for (var ni = 0; ni < this.scene.nodes.length; ni++) {
+            var firstNode = this.nodes[this.scene.nodes[ni]];
+            drawNodeRecursive(this, firstNode, mat4.create());
+        }
 
         // draw to the front buffer
         this.frontBuffer.drawImage(this.backBuffer, 0, 0);
     }
 
-    loadImage(imageInfo, gl) {
+    loadImage(imageInfo, gl,mesh) {
```

Comment:
Multiple meshes can use the same image/texture, and this implementation tries to avoid loading such images multiple times.  It would be cleaner for this code to work independently from a specific mesh, and instead have meshes reference the images they need.